compiler/x/codegen/OutlinedInstructions.cpp

```diff
@@ -175,18 +120,34 @@ void TR_OutlinedInstructions::generateOutlinedInstructionsDispatch()
    if (_targetReg)
       {
       TR_ASSERT(resultReg, "assertion failure");
+      TR_ASSERT(resultReg->getKind() == _targetReg->getKind(), "OutlinedInstructions: targetReg must have same time as resultReg.");
+
       TR::RegisterPair *targetRegPair = _targetReg->getRegisterPair();
       TR::RegisterPair *resultRegPair =  resultReg->getRegisterPair();
+
+      TR_ASSERT((targetRegPair == NULL && resultRegPair == NULL) ||
+                (targetRegPair != NULL && resultRegPair != NULL),
+                "OutlinedInstructions: targetReg and resultReg must be both register pairs or neither register pairs.");
       if (targetRegPair)
          {
-         TR_ASSERT(resultRegPair, "OutlinedInstructions: targetReg is a register pair and resultReg is not");
-         generateRegRegInstruction(_targetRegMovOpcode, _callNode, targetRegPair->getLowOrder(),  resultRegPair->getLowOrder(),  _cg);
-         generateRegRegInstruction(_targetRegMovOpcode, _callNode, targetRegPair->getHighOrder(), resultRegPair->getHighOrder(), _cg);
+         generateRegRegInstruction(MOVRegReg(), _callNode, targetRegPair->getLowOrder(),  resultRegPair->getLowOrder(),  _cg);
+         generateRegRegInstruction(MOVRegReg(), _callNode, targetRegPair->getHighOrder(), resultRegPair->getHighOrder(), _cg);
          }
       else
          {
-         TR_ASSERT(!resultRegPair, "OutlinedInstructions: resultReg is a register pair and targetReg is not");
-         generateRegRegInstruction(_targetRegMovOpcode, _callNode, _targetReg, resultReg, _cg);
+         switch (resultReg->getKind())
```

Comment:
This switch seems to not fall in either category of change you mentioned in the commit message. Could you please clarify what this code is for ?

Reply:
The previous version does not copy the result correctly for fcall/dcall/vcall, therefore I fix it.
In case if you are courious, the existing codes in OMR and known downstream projects do not use fcall/dcall/vcall with outlined instructions.

Reply:
Ok thanks for clarifying the reason (it's more than unused code cleanup).